pkg/executor/push.go

```diff
@@ -63,7 +63,15 @@ func CheckPushPermissions(opts *config.KanikoOptions) error {
 		if checked[destRef.Context().RepositoryStr()] {
 			continue
 		}
-		if err := remote.CheckPushPermission(destRef, creds.GetKeychain(), http.DefaultTransport); err != nil {
+
+		// Create a transport to set our user-agent.
+		tr := http.DefaultTransport
```

Comment:
Should there be (is there already?) some kind of `GetTransport()` method that does this in a shared location, sort of like how `creds.GetKeychain()` does? I'm not familiar with all of the available `opts`, and even if I were, if a new opt was added that configured HTTP requests like this, it'd be nice to have that applied in one common location.

I don't think it should necessarily block this PR, but it'd be nice to get feedback from someone more familiar with kaniko.